specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -72,7 +74,7 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
       this.fieldChangePromises[fieldName as string].resolve('superseded');
     this.fieldChangePromises[fieldName as string] = thisCheck;
 
-    const checks = [
+    const checks: Array<Promise<BusinessRuleResult<SCHEMA> | undefined>> = [
```

Comment:
for consistency, we don't use `Array` and `ReadonlyArray` anywhere in the codebase. Instead, we use `T[]` and `RA<T>`. feel free to change the convention, as long as you make all usages consistent

Reply:
Fixed in https://github.com/specify/specify7/pull/3792/commits/7a9d0b79016dad7adeb95f333e41479290935b13